### PR TITLE
[insights-core-3.0.300] Remove deprecated features

### PR DESCRIPTION
**Pull request by psachin**
Signed-off-by: Sachin Patil <psachin@redhat.com>

### All Pull Requests:

Check all that apply:

* [X] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

This PR will remove the deprecated features which are planned to remove after insights-core-3.0.300.

*Add your description here*

### CODE REVIEW

**Review by ryan-blakley — CHANGES_REQUESTED** [review]

File: insights/combiners/modinfo.py [hunk 1]
```diff
@@ -120,14 +42,14 @@ class ModulesInfo(dict):
...
-    def __init__(self, filtered_modules_info, i40e_info, igb_info, ixgbe_info, veth_info, vmxnet3_info):
+    def __init__(self, filtered_modules_info):
         self.retpoline_y = set()
         self.retpoline_n = set()
         if filtered_modules_info:
             self.update(filtered_modules_info)
             self.retpoline_n = filtered_modules_info.retpoline_n
             self.retpoline_y = filtered_modules_info.retpoline_y
-        for item in filter(None, [i40e_info, igb_info, ixgbe_info, veth_info, vmxnet3_info]):
+        for item in filter(None, []):
```
Comment (xiangce) [inline insights/combiners/modinfo.py:52]: Better remove this whole `for` loop?

**Review by xiangce — COMMENTED** [review]
@psachin - Per the discussion on the sync meeting, we suggested removing the RegisterPoints of the specs from `insights.specs.Specs` and their mapping in the relevant `insights.specs.DefaultSpecs` modules,  since after removing the parsers, these specs are not used elsewhere.

File: insights/combiners/modinfo.py (hunk 1 above)
Comment (psachin) [reply insights/combiners/modinfo.py:52]: Sure

File: insights/combiners/modinfo.py [hunk 2]
```diff
@@ -17,88 +17,10 @@
...
-
-    Attributes:
-        retpoline_y (set): A set of names of the modules with the attribute "retpoline: Y".
-        retpoline_n (set): A set of names of the modules with the attribute "retpoline: N".
-    """
-    def __init__(self, mi_all, mi_each):
-        deprecated(
-            ModInfo,
-            "Please use the :class:`insights.parsers.modinfo.KernelModulesInfo` instead.",
-            "3.1.25"
```
[the remark this answers is not in the document]
Comment (psachin) [reply insights/combiners/modinfo.py:71]: Thanks for pointing that out. As this depends on `ModInfoAll` & `ModInfoEach` parser, I might have to delay to removal of these parsers after 3.1.25 if that seems ok to you.

**Comment by psachin** [comment]
> @psachin - Per the discussion on the sync meeting, we suggested removing the RegisterPoints of the specs from `insights.specs.Specs` and their mapping in the relevant `insights.specs.DefaultSpecs/InsightsArchiveSpecs/SosSpecs` modules, since after removing the parsers, these specs are not used elsewhere.

Done

File: insights/combiners/modinfo.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryan-blakley) [reply insights/combiners/modinfo.py]: That's fine with me, whatever works best for you I'm good with.

**Review by xiangce — APPROVED** [review]

**Comment by xiangce** [comment]
Hey All, since this PR has the high potential to break the running of the existing rules. I'd like to merge it in advance before the release day, to leave ample reaction time for us whenever any breakage happens.

**Pull request merged**